### PR TITLE
Adds onBearerToken handler

### PR DESCRIPTION
**Pull request by creeonix**
onBearerToken handler is called each time we set Bearer Token.
Can be handy if you need to update token on 3rd party services when token is refreshed

### CODE REVIEW

File: lib/dispatch.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ export default class Dispatch {
     this.clientID = clientID;
     this.clientSecret = clientSecret;
     this.host = host;
+    this.onBearerToken = () => {};
```
Comment (derekcaneja) [inline lib/dispatch.js]: You should make this more reusable:
```
this.bearerTokenChangeHooks = [];
```

File: lib/dispatch.js [hunk 2]
```diff
@@ -176,6 +177,12 @@ export default class Dispatch {
     // Reset all aux clients for the same reason
     this.auxiliaryClients = {};
 
+    if (typeof this.onBearerToken === 'function') {
```
Comment (derekcaneja) [inline lib/dispatch.js]: Add a function above:
```
onBearerTokenChange(hook) {
  if (typeof hook === 'function') {
    this.bearerTokenChangeHooks.push(hook);
  } else {
    console.log('Dispatch SDK: onBearerTokenChange hook must be a function');
  }
}
```

Then change this to:
```
this.bearerTokenChangeHooks.forEach(hook => {
  hook(this);
});
```

**Comment by creeonix** [comment]
@derekcaneja now it supports multiple hooks.

**Comment by derekcaneja** [comment]
LGTM

**Pull request merged**